### ISSUE TITLE
CC membrane model missing pressure ratio limitation on achievable selectivity

### ISSUE THREAD

**Issue by jkitchin** [jkitchin/differentiable-flowsheets#144]
## Description

In src/difflow_cc/units/membrane.py (lines 247-253), the perfect-mixing selectivity equation does not account for the pressure ratio limitation. Even with infinite selectivity, the maximum permeate CO2 mole fraction is limited by the pressure ratio. The simplified equation used is only valid when the permeate-to-feed pressure ratio approaches zero. The correct equation includes the pressure ratio explicitly.

## Severity
MODERATE

## Component
CC Plugin - Membrane (src/difflow_cc/units/membrane.py)

**Comment by jkitchin** [comment]
## Fixed in commit 0f395a1

**Root cause:** In `MembraneSeparator.__call__()` in `src/difflow_cc/units/membrane.py`, the permeate CO2 mole fraction was computed purely from selectivity without considering the thermodynamic limit imposed by the pressure ratio. No membrane can achieve a permeate enrichment greater than `y_feed * (P_feed / P_permeate)` because that would require permeation against the partial pressure gradient.

**Fix:** Added a pressure ratio constraint on the permeate CO2 mole fraction:

```python
y_CO2_perm = alpha * y_CO2_feed / (1.0 + (alpha - 1.0) * y_CO2_feed)

# New: pressure ratio limits maximum permeate enrichment
pressure_ratio_val = P_feed / P_permeate
y_CO2_perm = jnp.minimum(y_CO2_perm, y_CO2_feed * pressure_ratio_val)
y_CO2_perm = jnp.clip(y_CO2_perm, 0.0, 0.999)
```

**Example:** A membrane with selectivity=100 at a pressure ratio of 10 with 4% CO2 feed: the selectivity-only formula gives y_perm = 0.807, but the pressure ratio limits it to y_perm = min(0.807, 0.04 * 10) = 0.4. The pressure ratio constraint correctly dominates.

**Test:** `tests/cc/test_membrane_bugs.py::test_membrane_pressure_ratio_limits_selectivity`
